mne/cov.py

```diff
@@ -76,17 +96,26 @@ class Covariance(dict):
         List of channels' names.
     nfree : int
         Number of degrees of freedom i.e. number of time points used.
+
+    See Also
+    --------
+    compute_covariance
+    compute_raw_data_covariance
+    make_ad_hoc_cov
+    read_cov
     """
 
-    def __init__(self, fname):
+    def __init__(self, data, names, bads, projs, nfree, eig=None, eigvec=None,
+                 method=None, loglik=None):
         """Init of covariance."""
-        if fname is None:
-            return
-
-        # Reading
-        fid, tree, _ = fiff_open(fname)
-        self.update(_read_cov(fid, tree, FIFF.FIFFV_MNE_NOISE_COV))
-        fid.close()
+        diag = True if data.ndim == 1 else False
+        self.update(data=data, dim=len(data), names=names, bads=bads,
+                    nfree=nfree, eig=eig, eigvec=eigvec, diag=diag,
+                    projs=projs, kind=FIFF.FIFFV_MNE_NOISE_COV)
```

Comment:
@agramfort those get set automatically here, so no consequence (other than cleaner/DRY code) to not setting them where you mention